app/views/spree/checkout/payment/_payu.html.erb

```diff
@@ -1 +1,5 @@
 <%= link_to Spree.t(:pay_with_payu), pay_with_payu_path(payment_method_id: payment_method.id) %>
+
+<script>
+  SpreePayuIntegration.paymentMethodID = "<%= payment_method.id %>"
```

Comment:
Also instead of passing some global JS values I would recommend taking it out from some kind of `data` attribute form HTML element (like this link which is generated in line 1).